apps/web/src/stores/latte/index.ts

```diff
@@ -0,0 +1,139 @@
+'use client'
+
+import { useCallback, useMemo } from 'react'
+import {
+  AppLocalStorage,
+  useLocalStorage,
+} from '@latitude-data/web-ui/hooks/useLocalStorage'
+import { Project } from '@latitude-data/core/browser'
+import { useLatteZustandStore } from './store'
+
+type StoredLatteData = { jobId?: string; threadUuid?: string }
+type StoredLatteDataKeys = keyof StoredLatteData
+export type StoredLatteDataByProject = Record<number, StoredLatteData>
+
+export const useLatteStore = ({ project }: { project: Project }) => {
+  const projectId = project.id
+  const { setValue: setStoredLatteData } =
+    useLocalStorage<StoredLatteDataByProject>({
+      key: AppLocalStorage.latteThread,
+      defaultValue: {},
+    })
+  const store = useLatteZustandStore()
+
+  const addOrRemoveValue = useCallback(
+    (key: StoredLatteDataKeys, value: string | undefined) => {
+      setStoredLatteData((prev) => {
+        if (value) {
+          return {
+            ...prev,
+            [projectId]: {
+              ...prev[projectId],
+              [key]: value,
+            },
+          }
+        } else {
+          const { [projectId]: projectData, ...rest } = prev
+          if (projectData) {
+            const { [key]: _, ...restProjectData } = projectData
+            return {
+              ...rest,
+              ...(Object.keys(restProjectData).length > 0
+                ? { [projectId]: restProjectData }
+                : {}),
+            }
+          }
+          return rest
+        }
+      })
+    },
+    [setStoredLatteData, projectId],
+  )
+
+  const saveInLocalStorage = useCallback(
+    (key: StoredLatteDataKeys, value: string | undefined) => {
+      setStoredLatteData((prev) => {
+        if (value) {
+          return {
+            ...prev,
+            [projectId]: {
+              ...prev[projectId],
+              [key]: value,
+            },
+          }
+        } else {
+          const { [projectId]: projectData, ...rest } = prev
+          if (projectData) {
+            const { [key]: _, ...restProjectData } = projectData
+            return {
+              ...rest,
+              ...(Object.keys(restProjectData).length > 0
+                ? { [projectId]: restProjectData }
+                : {}),
+            }
+          }
+          return rest
+        }
+      })
+    },
+    [setStoredLatteData, projectId],
+  )
+
+  const setThreadUuid = useCallback(
+    (uuid: string | undefined) => {
+      store.setThreadUuid(uuid)
+      addOrRemoveValue('threadUuid', uuid)
+      saveInLocalStorage('threadUuid', uuid)
+    },
+    [store, addOrRemoveValue, saveInLocalStorage],
+  )
+
+  const setJobId = useCallback(
+    (jobId: string | undefined) => {
+      store.setJobId(jobId)
+      addOrRemoveValue('jobId', jobId)
+      saveInLocalStorage('jobId', jobId)
```

Comment:
Same as before but now job/thread live under the same object:

```js
{ [projectId]: { jobId?: string; threadUuid?: string }}
```
As before we double save it in zustand and localStorage